File: psra/service/src/main/java/by/arhor/psra/mapper/impl/GalleryMapperImpl.java
package by.arhor.psra.mapper.impl;

import org.modelmapper.ModelMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import by.arhor.psra.dto.GalleryDto;
import by.arhor.psra.repository.model.Gallery;

@Component
public class GalleryMapperImpl extends AbstractMapper<Gallery, GalleryDto> {

	@Autowired
	public GalleryMapperImpl(ModelMapper mapper) {
		super(mapper);
	}

	@Override
	public GalleryDto mapToDto(Gallery entity) {
		return mapper.map(entity, GalleryDto.class);
	}

	@Override
	public Gallery mapToEntity(GalleryDto dto) {
		return mapper.map(dto, Gallery.class);
	}

}

File: psra/service/src/main/scala/by/arhor/psra/mapper/impl/GalleryMapperImpl.scala
package by.arhor.psra.mapper.impl

import by.arhor.psra.dto.GalleryDto
import by.arhor.psra.mapper.Mapper
import by.arhor.psra.repository.model.Gallery
import org.modelmapper.ModelMapper
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class GalleryMapperImpl @Autowired() (private val mapper: ModelMapper) extends Mapper[Gallery, GalleryDto] {

	override def mapToDto(entity: Gallery): GalleryDto = mapper.map(entity, classOf[GalleryDto])

	override def mapToEntity(dto: GalleryDto): Gallery = mapper.map(dto, classOf[Gallery])

}
